fvtest/compilertriltest/LongAndAsRotateTest.cpp

```diff
@@ -0,0 +1,494 @@
+std::vector<uint32_t> iu2lInput = {0x0,0x1,0x0000FFFF,0x2,0x3E,0x80000000,0xFFFF0000,0x40000000,0x7C000000,0xFFFFFFFF,0x0003C000,0x40000001};
+
+// This vector consists of all cases where contiguous 1's appear at most once in the binary value
+// ex: 001111000, 1111111, 011110, 1110000, 00001111
+std::vector<uint64_t> lconstInput_zerosAroundOnes =     {   0x0,
+                                                            0x1,                //00000..00001
+                                                            0x00000000FFFFFFFF, //00000..11111
+                                                            0x2,                //00000..00010
+                                                            0x3E,               //00...0111110     
+                                                            0x8000000000000000, //10000000..00
+                                                            0xFFFFFFFF00000000, //111111..0000
+                                                            0x4000000000000000, //010000..0000
+                                                            0x7C00000000000000, //0111100..000
+                                                            0xFFFFFFFFFFFFFFFF, //1111111..111
+                                                            0x00000003C0000000, //0..011110..0
+                                                            0x00007FFFFFFFFFFF,
+                                                            0x0000FFFFFFFFFFFF,
+                                                            0x000000000000FFFF,
+                                                            0xFFFFFFFFFFFE0000,
+                                                            0xFFFFFFFFFFFF0000,
+                                                            0xFFFF000000000000                     
+                                                        };
+
+// this vector is for tests where contiguous zeros only appear once in the binary value.
+// ex: 11100001111, 1000001, 0000000
+std::vector<uint64_t> lconstInput_onesAroundZeros = { 0x0, 0x8000000000000001, 0xE000000000000003, 0xFFFFFFF00FFFFFFF, 0xFFFFFFFEFFFFFFFF, 0xFFFFFFFE7FFFFFFF, 0xF0FFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFF0F, 0XFFFF000000000001, 0x8000FFFFFFFFFFFF };
+
+// These input values cannot be handled by a RISBG instruction
+std::vector<uint64_t> lconstInput_invalidPatterns = {0x4000000000000001, 0xF0F0F0F0F0F0F0F0, 0x8000000000000002};
+
+template <typename VarType, typename ConstType>
+using LongAndAsRotateParamType = std::tuple<VarType, ConstType, ConstType (*) (VarType, ConstType)>;
+
+template <typename VarType, typename ConstType>
+class LongAndAsRotateTest : public ::testing::TestWithParam<LongAndAsRotateParamType<VarType, ConstType>>
+   {
+   public:
+
+   static void SetUpTestCase()
+      {
+      const char *options = "-Xjit:disableTreeSimplification,disableAsyncCompilation,acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,useIlValidator,paranoidoptcheck";
```

Comment:
I wonder if this test case wouldn't be a good candidate to test out the MockOptimization strategies, just to give you some more direct control over the strategy. (See #1661 and [this example](https://github.com/mgaudet/omr/blob/3049c28624e9b989d289687e28a615cba201c502/fvtest/compilertriltest/SimplifierFoldAndTest.cpp#L69-L84))

Reply:
Interesting. I remember wishing there was something like this when I was testing edge cases. Some test cases were particularly designed at the time to test under different optimization settings. Not sure if it's needed now though, as we made some changes such that the testing was no longer needed. But I'll look into this.

Reply:
So I spoke to Filip about this and realized I need to rethink the optimization strategy for these tests. In most cases we should not be having any optimizations in the strategy because these trees will generally be folded by the optimizer. I will enable the trees in certain cases when I want the optimizer to detect certain patterns and set appropriate flags. Perhaps we can also revisit this when we have a feature in tril that allows us to set flags (as you suggested in another comment here). I'll open an issue relating to this today.

Reply:
So I played around with MockOptimization strategies, and I can make something work using that as well. However I still don't think this is the best long term solution. (Also, the segment of code I was trying to test will be removed in the next commit, but we'll ignore that for now as it can still be important to test this corner case).

The JIT code modified here is in the code generator and ideally I would like to run with noOpt to test the code. However I can't do that right now because some of the nodes require certain flags to be set. So my work around was to disable just enough optimizations so that the optimizer will set the flag I want. The way to do this with MockOptimization would be to run with noOpts except for a couple of test cases (where we would have to add then remove an optimization, or run these tests under a different TestFixture.

The best long term solution IMO here is to not use any optimization and set the flags myself. However the feature to set flags is not ready yet (https://github.com/eclipse/omr/issues/1947). I suggest we leave this as is and revisit these tests once we have the ability to set flags. Neither of the current solutions are particularly "cleaner" and are not the long term solution. So it doesn't really matter which one we go with right now. We will have to revisit this regardless. 

Let me know what you think.